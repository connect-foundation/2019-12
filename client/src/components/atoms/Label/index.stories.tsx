import React from 'react';
import { text, boolean } from '@storybook/addon-knobs';
import Label from '.';

export default {
  title: 'Atoms / Label',
};

export const defaultLabel: React.FC = () => (
  <Label required={boolean('required', false)} name={text('name', '이메일')} />
);

export const requiredLabel: React.FC = () => (
  <Label required={boolean('required', true)} name={text('name', '이메일')} />
);
