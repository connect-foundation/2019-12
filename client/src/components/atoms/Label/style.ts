import styled from 'styled-components';

export const Container = styled.label`
  /* display: inline-block; */
`;
export const NameSpan = styled.span`
  /* display: block; */
`;
export const AsteriskSpan = styled.span`
  display: inline-block;
  color: #FF2d54;
`;
