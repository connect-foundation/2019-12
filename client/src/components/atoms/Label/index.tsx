import React from 'react';
import * as S from './style';

export interface Props {
  name: string;
  required?: boolean;
}

function Label({ name, required = false }: Props): React.ReactElement {
  return <S.Label required={required}>{name}</S.Label>;
}

export default Label;
