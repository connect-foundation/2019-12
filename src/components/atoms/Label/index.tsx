import React from 'react';
import * as S from './style';

interface Props {
  name: string;
  required?: boolean;
}

function Label({ name, required = false }: Props): React.ReactElement {
  return <S.Label required={required}>{name}</S.Label>;
}

export default Label;
