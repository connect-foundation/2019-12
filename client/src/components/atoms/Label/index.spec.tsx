import React from 'react';
import { mount } from 'enzyme';
import Label from '.';

describe('Atom / Divider', () => {
  it('[SNAPSHOT] 렌더링', () => {
    // given
    const wrapper = mount(<Label required={false} name={'이메일'} />);

    // then
    expect(wrapper).toMatchSnapshot();
  });
});
