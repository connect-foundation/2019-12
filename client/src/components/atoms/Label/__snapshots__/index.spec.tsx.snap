// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom / Divider [SNAPSHOT] 렌더링 1`] = `
.c0::after {
  content: '*';
  visibility: hidden;
}

<Label
  name="이메일"
  required={false}
>
  <styled.label
    required={false}
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  ",
              [Function],
              "
  color: ",
              [Function],
              ";

  &::after {
    content: '*';
    visibility: ",
              [Function],
              ";
    color: ",
              [Function],
              ";
  }
",
            ],
          },
          "displayName": "styled.label",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "label",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      required={false}
    >
      <label
        className="c0"
        required={false}
      >
        이메일
      </label>
    </StyledComponent>
  </styled.label>
</Label>
`;
